### PR TITLE
Cancel all button

### PR DESCRIPTION
**Pull request by sircrusher**
- Added a cancel-all button to Perp Markets. Both desktop and mobile views.
- 1 issue where the initial transaction notification does not clear after the success notification.

### CODE REVIEW

File: components/OpenOrdersTable.tsx [hunk 1]
```diff
@@ -353,6 +377,43 @@ const OpenOrdersTable = () => {
...
+    let txid
+    try {
+      if (!selectedMangoGroup || !selectedMangoAccount || !wallet) return            
+      if (market instanceof PerpMarket) {             
+        txid = await mangoClient.cancelAllPerpOrders(
+          selectedMangoGroup,
+          [market],          
+          selectedMangoAccount,
+          wallet.adapter
+        )
```
Comment (tjshipe) [inline components/OpenOrdersTable.tsx:399]: it looks like `cancelAllPerpOrders` returns an array of txids

you can loop over the array of txids to dismiss the pending notifications. here is an example: https://github.com/blockworks-foundation/mango-ui-v3/blob/main/components/AccountInfo.tsx#L102

**Review by tjshipe — CHANGES_REQUESTED** [review]

File: components/OpenOrdersTable.tsx (hunk 1 above)
Comment (sircrusher) [reply components/OpenOrdersTable.tsx]: you were right. this fixed it.

**Comment by tjshipe** [comment]
Approved. Pinging @saml33 for a review on this one to double check design before merging.

**Review by tjshipe — APPROVED** [review]

**Comment by saml33** [comment]
Can we have it so the button persists when you view other perp markets? The Cancel All button could be inline with the first order for that market (only show if there is more than one order for the same market). We might need to change the styling of the Edit/Cancel buttons to make it work but I can help with that if you want.

**Comment by sircrusher** [comment]
> 

The button currently does persist on other perp markets, as long as there is at least 1 open order for the market you are currently viewing. I have tagged you on a comment for the line that determines this for the desktop view.

If I have misunderstood your request, please let me know 🤙

**Comment by saml33** [comment]
@sircrusher yeah, I mean if I'm viewing BTC-PERP and I have multiple SOL-PERP orders (which I can see in the table) it might be nice to be able to cancel all of the SOL-PERP orders from the BTC-PERP page.

The first order of a market with multiple orders could have the cancel all button beside the cancel button

**Comment by sircrusher** [comment]
I moved the button to the orders row and made it red.

Button shows up on every perp order of the same contract if there are 2 or more orders for the same contract.

Desktop and mobile screenshots below:

image
 and
image

**Comment by sircrusher** [comment]
I can't see any details on the failed check. Get 404 when I try to view them.

**Comment by godmodegalactus** [comment]
We can update this PR to include cancel all spot orders too.
@sircrusher should I update it?

**Comment by sircrusher** [comment]
> We can update this PR to include cancel all spot orders too. @sircrusher should I update it?

sure, go ahead :)

**Pull request merged**